app/services/background.py

```
@@ -375,52 +375,231 @@ async def _market_data_sync_service(self):
+                    self.logger.warning(f"Discovery strategy failed: {strategy['description']}", error=str(e))
+                    continue
+            
+        except Exception as e:
+            self.logger.error("Market analysis discovery failed", error=str(e))
+        
+        # ENTERPRISE: Direct exchange API discovery (bypass service layer if needed)
+        try:
+            exchange_symbols = await self._discover_symbols_direct_apis()
+            all_discovered_symbols.update(exchange_symbols)
+            self.logger.info(f"Direct API discovery added {len(exchange_symbols)} additional symbols")
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)